### DISCUSSION TITLE
Pool Temp vs set point

### DISCUSSION THREAD

**Discussion by tblegen** [tagyoureit/nodejs-poolController#877] [Nixie]
New to njsPC, not sure if I have configured correctly or just missing something. I have a 12k gallon indoor pool, with Intellifllo VS pump, and Mastertemp 250 heater. Both connected via RS485 and I have working controls. I added a 10k temperature probe on the I2c bus for water temperature.
My issue is with the heater control. I have the Set Point at 84F, but the heater keeps cycling with only a 0.1 degree change in water temp. this causes the heater to cycle every few seconds. Is there a way to adjust the cycling rate or configure the system to respond to say only full degree changes to the temp probe. 
I'm using the nixie single body setup. I do have it connected to Home Assistant with the njsPC-HA integration. Everything else is working good except for the heater short cycling. I have also disabled the HA portion to remove that from the equation. Any help would be appreciated. Thanks

**Comment by rstrouse** [comment]
Are you running your MasterTemp on RS485?  The gas relay has heuristics to seek the temp and stop short cycles with a timing mechanism.  The RS485 control is up to the heater to perform the call for heat.  If you are using a heater control relay check the settings for your heater to ensure you have set a minimum cycle time.

**Comment by tblegen** [comment]
I am running the Master Temp on RS485. The MasterTemp is configured with a min and max pool temperature. nodejsPC sets the MasterTemp to the correct temperature (set point). All is well until the "Pool Temp" reaches the nodejsPC Set Point.  nodejsPC then cycles the "Heater Status" between Off and Heater based on the nodejsPC "Pool Temp", with as little as a 0.1 degree change above or below the nodejsPC "Set Point". This causes the MasterTemp to cycle off and on, as if I would manually operate the "Heater Off" or "Pool On" switch.
If I understand you correctly, nodejsPC should configure the Set Point on the MasterTemp, then stay out of the way. Instead it operates more like an on/off switch for the heater. 
I hope I stated this a little better than my original post. I apologize for any confusion and thank you for your help.

**Reply by rstrouse** [reply]
This operates just like it does with IntelliCenter.   However, this doesn't make it right.   The gas heater functions are much more advanced and perform heuristics so in this sense the fireman's switch is much smarter.  I'll have a look and see if they can be added to the MasterTemp RS485 as well.  The big difference with IntelliCenter is that the temperature is only recorded in 1 degree increments but we can do better by adapting the heuristics that exist for solar and gas heater have, to the MasterTemp code and holding the call for heat if it shut off with the minimum cycle time.

In the mean time you should set the water temp to take median samples so that it does not rapidly reflect the water temperature.

**Comment by rstrouse** [comment]
If you pull the latest njsPC and dashPanel there are two new options for MasterTemp heaters.  These include the Stop Temp Delta which will tell the heater to heat past the setpoint if needed by x degrees and another option to identify the minimum cycle time that the heater should fire for.  This should handle your short cycles but I do not have a MasterTemp on RS485 to try it out.

**Comment by tblegen** [comment]
Thank you for looking at this. I'll give this a try and let you know.

**Comment by tblegen** [comment]
I installed your new version. Now, the heater is off for ~20 min. and on for ~5. This is with both the "stop temp delta" and "min cycle time" set to 2 minutes. Huge improvement over the ~10 second cycles I was seeing before. Thank you for providing this patch.
